src/Bicep.Core.UnitTests/Assertions/CompilationAssertions.cs

```diff
@@ -27,23 +29,29 @@ public CompilationAssertions(Compilation compilation)
 
         public AndConstraint<CompilationAssertions> ContainDiagnostic(string code, DiagnosticLevel level, string message, string because = "", params object[] becauseArgs)
         {
-            Subject.GetEntrypointSemanticModel().GetAllDiagnostics().Should().ContainDiagnostic(code, level, message, because, becauseArgs);
+            var noLinterConfig = new ConfigHelper().GetDisabledLinterConfig();
```

Comment:
I think this is something we should revisit (worth creating an issue?). It feels a little unexpected that linting should be disabled by default in diagnostic assertions. I'm wondering whether we should have separate methods for accessing 'core compiler' & linter diagnostics.